data/debug-cli/docker_debug.yaml

```diff
@@ -0,0 +1,318 @@
+  ```
+  
+  To confirm nginx is running, open a browser and navigate to http://localhost:8080.
+  You should see the default nginx page.
+  Let's change it using vim:
+  
+  ```console
+  vim /usr/share/nginx/html/index.html
+  ```
+  
+  E.g., change the title to "Welcome to my app!" and save the file.
+  Now, reload the page in the browser and you should see the updated page.
+  
+  ### Managing your toolbox using the `install` command
+  
+  The builtin `install` command allows you to add any tool from https://search.nixos.org/packages to the toolbox.
+  Keep in mind adding a tool never modifies the actual image or container.
+  Tools get just added to your toolbox.
+  E.g., let's install `nmap`:
+  
+  ```console
+  $ docker debug nginx
+  ...
+  docker > install nmap
+  Tip: You can install any package available at: https://search.nixos.org/packages.
+  installing 'nmap-7.93'
+  these 2 paths will be fetched (5.58 MiB download, 26.27 MiB unpacked):
+  /nix/store/brqjf4i23fagizaq2gn4d6z0f406d0kg-lua-5.3.6
+  /nix/store/xqd17rhgmn6pg85a3g18yqxpcya6d06r-nmap-7.93
+  copying path '/nix/store/brqjf4i23fagizaq2gn4d6z0f406d0kg-lua-5.3.6' from 'https://cache.nixos.org'...
+  copying path '/nix/store/xqd17rhgmn6pg85a3g18yqxpcya6d06r-nmap-7.93' from 'https://cache.nixos.org'...
+  building '/nix/store/k8xw5wwarh8dc1dvh5zx8rlwamxfsk3d-user-environment.drv'...
+  
+  docker > nmap --version
+  Nmap version 7.93 ( https://nmap.org )
+  Platform: x86_64-unknown-linux-gnu
+  Compiled with: liblua-5.3.6 openssl-3.0.11 libssh2-1.11.0 nmap-libz-1.2.12 libpcre-8.45 libpcap-1.10.4 nmap-libdnet-1.12 ipv6
+  Compiled without:
+  Available nsock engines: epoll poll select
+  ```
+  
+  You can confirm `nmap` is now part of your toolbox, e.g., by getting a debug shell into a different image:
+  
+  ```console
+  $ docker debug hello-world
+  ...
+  docker > nmap --version
+  
+  Nmap version 7.93 ( https://nmap.org )
+  Platform: x86_64-unknown-linux-gnu
+  Compiled with: liblua-5.3.6 openssl-3.0.11 libssh2-1.11.0 nmap-libz-1.2.12 libpcre-8.45 libpcap-1.10.4 nmap-libdnet-1.12 ipv6
+  Compiled without:
+  Available nsock engines: epoll poll select
+  
+  docker > exit
+  ```
+  
+  `nmap` is still there.
+  
+  
+  ### Understanding the default startup command of a container (entry points)
+  
+  Docker debug comes with a builtin tool `entrypoint`.
+  E.g., if we enter the `hello-world` image, we can confirm the entrypoint is `/hello`:
+  
+  ```console
+  $ docker debug hello-world
+  ...
+  docker > entrypoint --print
+  /hello
+  ```
+  
+  The `entrypoint` command evaluates the `ENTRYPOINT` and `CMD` statement of the underlying image 
+  and allows to print, lint, or run the resulting actual entrypoint.
+  However, keeping all the corner cases of https://docs.docker.com/engine/reference/builder/#understand-how-cmd-and-entrypoint-interact in mind can be tricky so `entrypoint` can be a great help in many situations.
```

Comment:
```suggestion
  However, it can be difficult to understand all the corner cases from [Understand how CMD and ENTRYPOINT interact](/engine/reference/builder/#understand-how-cmd-and-entrypoint-interact). In these situations, `entrypoint` can help.
```